torch/autograd/gradcheck.py

```diff
@@ -305,6 +305,23 @@ def jvp_fn(delta):
     return jvp_fn
 
 
+def reshape_tensor_or_tuple(u, shape):
+    if isinstance(u, tuple):
+        if u[0].layout != torch.sparse_coo:
+            return (u[0].reshape(shape), u[1].reshape(shape))
+    else:
+        if u.layout != torch.sparse_coo:
+            return u.reshape(shape)
+    return u
```

Comment:
nit: add a comment here saying that we don't need to reshape for sparse Tensors.